hornet-private-net/extra-nodes/private-hornet.sh

```diff
@@ -0,0 +1,323 @@
+
+if [ -z "$api_port" ]; then
+  api_port=$DEFAULT_API_PORT
+fi
+
+if [ -z "$peering_port" ]; then
+  peering_port=$DEFAULT_PEERING_PORT
+fi
+
+if [ -z "$dashboard_port" ]; then
+  dashboard_port=$DEFAULT_DASHBOARD_PORT
+fi
+
+if [ -n "$3" ]; then
+  coo_public_key="$3"
+else 
+  if [ -f ../coo-milestones-public-key.txt ]; then
+    coo_public_key=$(cat ../coo-milestones-public-key.txt | tr -d "\n")
+  else 
+    echo "Please provide the coordinator's public key"
+    exit 131
+  fi
+fi
+
+if [ -n "$4" ]; then
+  peer_address="$4"
+else 
+  if [ -f ../node1.identity.txt ]; then
+    peer_address="/dns/node1/tcp/15600/p2p/$(getPeerID ../node1.identity.txt)"
+  else
+    echo "Please provide a peering address"
+    exit 132
+  fi
+fi
+
+if [ -n "$5" ]; then
+  snapshot_file="$5"
+else 
+  if [ -f ../snapshots/private-tangle/full_snapshot.bin ]; then
+    snapshot_file="../../../snapshots/private-tangle/full_snapshot.bin"
+  else
+    echo "Please provide a snapshot file of your Private Tangle"
+    exit 133
+  fi
+fi
+
+
+# Basic bootstrapping of folders for our Node
+
+if ! [ -d ./nodes ]; then
+  mkdir ./nodes
+fi
+
+if ! [ -d ./nodes/"$node_name" ]; then
+  mkdir ./nodes/"$node_name"
+fi
+
+cd  ./nodes/"$node_name"
+
+clean () {
+  stopContainers
+
+  if [ -d ./db ]; then
+    sudo rm -Rf ./db/*
+  fi
+
+  if [ -d ./p2pstore ]; then
+    sudo rm -Rf ./p2pstore
+  fi
+
+  if [ -d ./config ]; then
+    sudo rm -Rf ./config/*
+  fi
+
+  if [ -d ./snapshots/private-tangle ]; then
+    sudo rm -Rf ./snapshots/private-tangle/*
+  fi
+}
+
+# Sets up the necessary directories if they do not exist yet
+volumeSetup () {  
+  if ! [ -d ./config ]; then
+    mkdir ./config
+  fi
+
+  if ! [ -d ./db ]; then
+    mkdir ./db
+  fi
+
+  # P2P
+  if ! [ -d ./p2pstore ]; then
+    mkdir ./p2pstore
+  fi
+
+  if ! [ -d ./snapshots ]; then
+    mkdir ./snapshots
+  fi
+
+  if ! [ -d ./snapshots/private-tangle ]; then
+    mkdir ./snapshots/private-tangle
+  fi
+
+  ## Change permissions so that the Tangle data can be written (hornet user)
+  ## TODO: Check why on MacOS this cause permission problems
+  if ! [[ "$OSTYPE" == "darwin"* ]]; then
+    echo "Setting permissions for Hornet..."
+    sudo chown -R 65532:65532 ./db 
+    sudo chown -R 65532:65532 ./p2pstore
+    sudo chown -R 65532:65532 ./snapshots
+  fi 
+}
+
+bootstrapFiles () {
+  cp ../../docker-compose.yml .
+  sed -i 's/node/'$node_name'/g' docker-compose.yml
+  sed -i 's/0.0.0.0:'$DEFAULT_API_PORT'/0.0.0.0:'$api_port'/g' docker-compose.yml
+  sed -i 's/0.0.0.0:'$DEFAULT_PEERING_PORT'/0.0.0.0:'$peering_port'/g' docker-compose.yml
+  sed -i 's/0.0.0.0:'$DEFAULT_DASHBOARD_PORT'/0.0.0.0:'$dashboard_port'/g' docker-compose.yml
+
+  cp ../../../config/config-node.json ./config/config.json
+  sed -i 's/node1/'$node_name'/g' ./config/config.json
+
+  cp ../../../config/profiles.json ./config/profiles.json
+  cp ../../peering.json ./config/peering.json
+
+  cp "$snapshot_file" ./snapshots/private-tangle
```

Comment:
Copying over the snapshot file, required sudo priviledge. The code had to be edited to

```
sudo cp "$snapshot_file" ./snapshots/private-tangle
```